iaf-commons/src/main/java/nl/nn/adapterframework/util/YamlParser.java

```diff
@@ -0,0 +1,114 @@
+/*
+   Copyright 2023 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.util;
+
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.yaml.snakeyaml.Yaml;
+
+public class YamlParser extends Properties {
+
+	/**
+	 * Loads and parses a yaml file.
+	 * Uses the {@link #handleRawValue(String, Object)} method to recursively loop over all values.
+	 */
+	public YamlParser(Reader reader) {
+		Yaml yaml = new Yaml();
+
+		Map<String, Object> obj = yaml.loadAs(reader, Map.class);
+		obj.entrySet().forEach(entry -> handleRawValue(entry.getKey(), entry.getValue()) );
```

Comment:
As I mentioned above, I would prefer not to put the loading inside the constructor.

I would rather have a separate `parse()` / `load()` method which returns a `Map` (or `Properties`) instance.